src/impl/seismic/kernels/SurfIntFused.hpp

```diff
@@ -382,6 +388,51 @@ class edge::seismic::kernels::SurfIntFused: public edge::seismic::kernels::SurfI
       }
     }
 
+    inline void exec_mm_kernel( unsigned short     i_group, 
+                                unsigned short     i_kernel,
+                                const TL_T_REAL *a,
+                                const TL_T_REAL *b,
+                                TL_T_REAL *c ) {
+#ifdef PP_MMKERNEL_PERF
+#ifdef _OPENMP
+      unsigned int tid = omp_get_thread_num();
+#else
+      unsigned int tid = 0;
+#endif
+      size_t start = _rdtsc();
+#endif
+      m_mm.m_kernels[i_group][i_kernel]( a, b, c );
+#ifdef PP_MMKERNEL_PERF
+      size_t duration = _rdtsc() - start;
+      m_mm.m_kernel_stats[tid][i_group][i_kernel].invocations++;
+      m_mm.m_kernel_stats[tid][i_group][i_kernel].cycles += duration; 
+#endif
+    }
+
+    inline void exec_mm_kernel_pf( const unsigned short    i_group, 
+                                  const unsigned short     i_kernel,
+                                  const TL_T_REAL *a,
+                                  const TL_T_REAL *b,
+                                        TL_T_REAL *c,
+                                  const TL_T_REAL *a_pf,
+                                  const TL_T_REAL *b_pf,
+                                  const TL_T_REAL *c_pf  ) {
```

Comment:
Add respective underscores `i_a`, [...], `io_c`; fix indentation, rename function (no underscores).
Wouldn't it be better to move this wrapper to the kernels themselves?

Reply:
it should be in this fine such that the compiler could inline. In fact a macro might be the best....

Reply:
Ok; inline works.